options.go

```diff
@@ -41,10 +40,7 @@ func WithDelLinkFunc(inFunc func(core.LinkDefinition) error) ProviderOption {
 			if err != nil {
 				return err
 			}
-			// shutdown specific NATs subscription
-			wp.natsSubscriptions[linkdef.ActorId].Drain()
-			wp.natsSubscriptions[linkdef.ActorId].Unsubscribe()
-			
+
 			wp.deleteLink(linkdef)
```

Comment:
I see where you have moved the deleting of the links to the deleteLink function, but where have you moved the Drain/Unsubscribe call to make sure inflight calls are completed?  

Also, I think we ONLY need the Drain, I think I was being overly eager here

Reply:
When a link is deleted, There is no need to Drain the subscription. In favor of the `validateProviderInvocation` function added in the previous pr, the new requests from the actor related to the delete link will be rejected, while these in-flight calls proceed as normal.

As for the Drain of this subscription, In rust SDK the provider subscribes to this topic for all actors calling the provider, so this subscription Drain when `ShutdownFunc` is called.

Reply:
Give me a day or two to think about this.  I feel as if we shouldn't be maintaining subscriptions on the lattice for no reason and I'm not completely sure what would happen if I re-linked with a different value map.

Reply:
Drain is the biggest one here. That makes sure all inflight messages are set out to the server insofar as I know. We should definitely drain if we can.

Reply:
After speaking with @autodidaddict , I agree with Taylor...we need to maintain the drain.

Reply:
@thomastaylor312 @jordan-rash Thanks for all your review and reply. I am not a nats expert and am not sure if or where the drain is needed. So I add the explanation of why I just deleted the drain here, to see if there is something I missed or misunderstood.
1. the Provider listens to the nats topic according to its public key when it starts or has a link put for the first time: `wasmbus.rpc.${LatticeRpcPrefix}.${ProviderKey}.${LinkName}`, all actors linked to this provider will send a message to this topic when they call the provider.
2. When an actor creates a link with this provider, the provider sdk will record the link information (when receiving the message, it will determine whether to allow the actor to call based on whether the actor is in the record) and call back the corresponding callback function of the user.
3. So when an actor's link with this provider is deleted, the provider just deletes the record and then calls back the user's corresponding callback function, without closing the topic listened to in step 1 in the process. I see that in the rust code is handled in this way https://github.com/wasmCloud/weld/blob/main/rpc-rs/src/provider.rs#L645-L649
```
            if let Some(ld) = &this.parse_msg::<LinkDefinition>(&msg, "link.del") {
                this.delete_link(&ld.actor_id).instrument(span.clone()).await;
                // notify provider that link is deleted
                provider.delete_link(&ld.actor_id).instrument(span).await;
            }
```
Besides according to the [nats drain doc](https://pkg.go.dev/github.com/nats-io/nats.go#Conn.Drain), it seems that once drain is called the topic is unsubscribed and the conn will be closed. But instead of using separate topic for every actor, the provider uses one topic to receive all actor call messages, so this topic should not be closed while one link delete.